echopype/mask/api.py

```diff
@@ -17,6 +17,231 @@
 }
 
 
+def validate_and_collect_mask_input(
+    mask: Union[
+        Union[xr.DataArray, str, pathlib.Path], List[Union[xr.DataArray, str, pathlib.Path]]
+    ],
+    storage_options_mask: Union[dict, List[dict]],
+) -> Union[xr.DataArray, List[xr.DataArray]]:
+    """
+    Validate that the input ``mask`` and associated ``storage_options_mask`` are correctly
+    provided to ``apply_mask``. Additionally, form the mask input that should be used
+    in the core routine of ``apply_mask``.
+
+    Parameters
+    ----------
+    mask: xr.DataArray or str or pathlib.Path or list of xr.DataArray or str or pathlib.Path
+        The mask(s) to apply to the variable specified by ``var_name``. This input can be a
+        single input or list that corresponds to a DataArray or a path to a DataArray. If a path
+        is provided this should point to a zarr or netcdf file with only one array in it.
```

Comment:
```suggestion
    mask: xr.DataArray, str, pathlib.Path, or a list of these datatypes
        The mask(s) to be applied. 
        Can be a single input or list that corresponds to a DataArray or a path. If a path
        is provided this should point to a zarr or netcdf file with only one data variable in it.
```
Changing the wording to reflect the scope of this function and clarify terminology.